src/server/json_family.cc

```diff
@@ -915,19 +934,11 @@ auto OpArrPop(const OpArgs& op_args, string_view key, WrappedJsonPath& path, int
       return {};
     }
 
-    size_t removal_index;
-    if (index < 0) {
-      int temp_index = index + val->size();
-      removal_index = abs(temp_index);
-    } else {
-      removal_index = index;
-    }
+    size_t array_size = val->size();
+    size_t removal_index = std::min(GetJsonArrayIndex(index, array_size), array_size - 1);
```

Comment:
Why do we need this `min()` call? In which situation will `array_size - ` be returned?

Reply:
Sorry, I didn't understand what `array_size - ` means. But we need `std::min` for the case when the index specified by the user is greater than the length of the array. For example, if the array has a length of 5 and the user specifies 15, the normalized index should be 4.

Reply:
Sorry, I meant to write `array_size - 1`, but you answered my question :upside_down_face: 
--
So that won't work, right?
If size is 15, index is 5, `NormalizeIndex()` would return `0`, and so the result of `std::min(0, 4)` would still be `0`. Am I missing something?

Reply:
Why `NormalizeIndex()` would return 0? `NormalizeIndex` does not change positive indexes, see [here](https://github.com/dragonflydb/dragonfly/pull/3712/files#diff-0799c8810dbf8758f9da40162075f38bb51639f806be28a06d9f669a6ebb0f76R273). So in this case the result of `std::min(5, 14)` would be 5.

Reply:
You're right. My bad.
Should it change positive indices though? I'll let you decide